static/app/views/dashboards/widgetCard/metricWidgetCard/index.tsx

```diff
@@ -183,32 +199,91 @@ type MetricWidgetChartContainerProps = {
   widget: Widget;
   dashboardFilters?: DashboardFilters;
   editorParams?: Partial<MetricWidgetQueryParams>;
+  renderErrorMessage?: (errorMessage?: string) => React.ReactNode;
 };
 
 export function MetricWidgetChartContainer({
   selection,
   widget,
   editorParams = {},
   dashboardFilters,
+  renderErrorMessage,
 }: MetricWidgetChartContainerProps) {
   const metricWidgetQueryParams = {
     ...convertFromWidget(widget),
     ...editorParams,
   };
 
+  const {projects, environments, datetime} = selection;
+  const {mri, op, groupBy, displayType} = metricWidgetQueryParams;
+
+  const {
+    data: timeseriesData,
+    isLoading,
+    isError,
+    error,
+  } = useMetricsDataZoom(
+    {
+      mri,
+      op,
+      query: extendQuery(metricWidgetQueryParams.query, dashboardFilters),
+      groupBy,
+      projects,
+      environments,
+      datetime,
+    },
+    {fidelity: displayType === MetricDisplayType.BAR ? 'low' : 'high'}
+  );
+
+  const chartRef = useRef<ReactEchartsRef>(null);
+
+  const chartSeries = useMemo(() => {
+    return timeseriesData
+      ? getChartTimeseries(timeseriesData, {
+          getChartPalette: createChartPalette,
+          mri,
+          groupBy,
+        })
+      : [];
+  }, [timeseriesData, mri, groupBy]);
+
+  if (isError) {
+    const errorMessage =
+      error?.responseJSON?.detail?.toString() || t('Error while fetching metrics data');
+    return (
+      <Fragment>
+        {renderErrorMessage?.(errorMessage)}
+        <ErrorPanel>
+          <IconWarning color="gray500" size="lg" />
+        </ErrorPanel>
+      </Fragment>
+    );
+  }
+
+  if (timeseriesData?.groups.length === 0) {
+    return (
+      <EmptyMessage
+        icon={<IconSearch size="xxl" />}
+        title={t('No results')}
+        description={t('No results found for the given query')}
+      />
+    );
+  }
+
   return (
-    <MetricWidgetBody
-      widgetIndex={0}
-      datetime={selection.datetime}
-      projects={selection.projects}
-      environments={selection.environments}
-      mri={metricWidgetQueryParams.mri}
-      op={metricWidgetQueryParams.op}
-      query={extendQuery(metricWidgetQueryParams.query, dashboardFilters)}
-      groupBy={metricWidgetQueryParams.groupBy}
-      displayType={toMetricDisplayType(metricWidgetQueryParams.displayType)}
-      chartGroup={DASHBOARD_CHART_GROUP}
-    />
+    <MetricWidgetChartWrapper>
+      <TransitionChart loading={isLoading} reloading={isLoading}>
+        <LoadingScreen loading={isLoading} />
+        <MetricChart
+          ref={chartRef}
+          series={chartSeries}
+          displayType={displayType}
+          operation={op}
+          widgetIndex={0}
```

Comment:
Yeah, we should move the `useFocusArea` out of the chart 😅 (not in this PR but rather soonish)